src/tasks/block.rs

```diff
@@ -83,11 +83,9 @@ impl InProgressBlock {
 
         if let Ok(txs) = txs {
             self.unseal();
-            // extend the transactions with the decoded transactions.
-            // As this builder does not provide bundles landing "top of block", its fine to just extend.
```

Comment:
I would not remove these comments for now as we're still doing what the comment says, and serves as a reminder that this should probably change (else i'd also link the ticket about how to deal with bundles that i suggested below)